### DISCUSSION TITLE
A Freestanding Rust Binary (ru)

### DISCUSSION THREAD

**Discussion by giscus** [phil-opp/blog_os#1102] [Post Comments (translated)]
# A Freestanding Rust Binary (ru)

Первый шаг в создании собственного ядра операционной системы — это создание исполняемого файла на Rust, который не будет подключать стандартную библио…

https://os.phil-opp.com/ru/freestanding-rust-binary/

**Comment by xierongchuan** [comment]
This is cool! :)

**Comment by master30f** [comment]
Am I the only one having issues with rust-analyzer marking the panic handler fn as a duplicate originally defined in std, even though i have the no std attrib at the beginning of the file? (the code still builds fine though)

**Reply by bjorn3** [reply]
Setting the `rust-analyzer.checkOnSave.allTargets` config to false should fix that issue. Once you get to the testing chapter I think you can re-enable it. Basically the issue is that rust-analyzer by default does `cargo check --all-targets` which causes not just the main executable to be checked, but also to be built as test executable. This requires the test library from the sysroot which pulls std. Std defines a panic handler too, but isn't used by blog os as it doesn't work without running on top of an os supported by std.

**Comment by Cerber-Ursi** [comment]
Just run into [this question](https://stackoverflow.com/questions/72508578/cargo-toml-dependencies-are-ignored/72509192#72509192) on SO, so I'd like to ask - is it intentional that Cargo.toml here doesn't have any `edition` field?

**Reply by bjorn3** [reply]
I don't think so. Looks like the same error was made in the original english version.

**Reply by Cerber-Ursi** [reply]
Whoops. I thought I've made this comment under the original, sorry for the confusion.
